packages/ng/src/app/select/select.component.ts

```diff
@@ -271,13 +283,17 @@ implements ControlValueAccessor, AfterContentInit, OnInit, OnDestroy {
 	@HostListener('blur', ['$event'])
 	blur(e) {
 		this._onTouched();
+		if (this.optionFeeder && this.optionFeeder.hasFocus()) {
+			return;
+		}
```

Comment:
?

Reply:
Same remarq, when you click in the `input` field of the search component, as the search component is place in a OverLay (out of the current component), Angular consider that blur is fire. But in fact, it false, it's an inner component that have generate the `blur` event... So the only way to deal with that is to ask the inner component if he has the focus

Reply:
ok but why is `hasFocus()` a function ?

Reply:
you can use a `getter` and be donne with it

Reply:
It's a function because, lot's of things could define that you have a focus. Imagine a multiple search field in top of the popup with an `input` and `radio button`. I know it's not the best choice for UI but it could be done, so `hasFocus` isn't resume to a single field but to multiples and to maybe a choice.

And the pb with getter, is for me that it's a concept link to concrete class and not to interfaces

Reply:
what angular does with form controls is they have a lot of getters
[FormControl](https://angular.io/api/forms/FormControl) - [AbstractControl](https://angular.io/api/forms/AbstractControl)

they do stuff i dont like like 
 - calling something `AbstractStuff` and not making it `abstract`
 - not using interfaces :(

wot i mean is you can have this architecture

```ts
export interface IOptionFeeder {
  focused: boolean; // not a function
}
export abstract class AOptionFeeder implementsIOptionFeeder {
  protected _focused: boolean;
  get focused() { return focused; }
}
export class MyOptionFeeder extends AOptionFeeder {
  // here you handle events and set this._focused when the feeder is focused
}
```

Reply:
defining interfaces is the first step to follow the [interface segregation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle), and if you do it well it's supes dupes ez to respect the [open/close principle](https://en.wikipedia.org/wiki/Open/closed_principle)